p2p-sync/src/main/java/org/smartregister/p2p/activity/P2pModeSelectActivity.java

```diff
@@ -285,7 +409,8 @@ public Context getContext() {
 
     @Override
     public void initializePresenter() {
-        presenter = new P2pModeSelectPresenter(this, new P2pModeSelectInteractor(this));
+        interactor = new P2pModeSelectInteractor(this);
```

Comment:
Should the view know about the interactor or have access to the interactor?

Reply:
Should we strictly perform such interactions through the presenter?

Removing the interactor interactions from the view is a small change. I only had one issue with moving the instantiation of the interactor to the presenter. It would bring in the actual interactor implementation in the presenter which is supposed to be free from that for easy mocking and testing. What do you think?